[WIP] `Changelog` - Content Design (#1706)
hashicorp/design-system: website/docs/whats-new/release-notes/partials/components.md

 **`npm package: @hashicorp/design-system-components`**

## 2.12.2 <Doc::Badge @type="outlined">Latest</Doc::Badge>

**September 22nd, 2023**

> - `PowerSelect` fix style overrides when the list is positioned above. [(#1655)](https://github.com/hashicorp/design-system/pull/1655) Thanks [@alex-ju](https://github.com/alex-ju)! 
> 
> - `Text` fixed issue with whitespace adding extra underline when used in links (eg. inside a `Link::Inline`)[(#1652)](https://github.com/hashicorp/design-system/pull/1652) Thanks [@didoo](https://github.com/didoo)! 
> 
> - 🔄 Updated dependencies:
>     - @hashicorp/ember-flight-icons@3.1.3

Reviewer: ```suggestion
- `PowerSelect` - fixed the style overrides when the list is positioned above. [#1655](https://github.com/hashicorp/design-system/pull/1655) Thanks [@alex-ju](https://github.com/alex-ju)! 

- `Text` - fixed an issue with whitespace adding an extra underline when used in links (e.g., inside a `Link::Inline`)[#1652](https://github.com/hashicorp/design-system/pull/1652) Thanks [@didoo](https://github.com/didoo)! 

- 🔄 Updated dependencies:
    - @hashicorp/ember-flight-icons@3.1.3
```

## 2.12.1

**September 19th, 2023**

> - `Pagination` bugfix `aria-label` on the component. [(#1627)](https://github.com/hashicorp/design-system/pull/1627) Thanks [@MelSumner](https://github.com/MelSumner)! 
> 
> - `Dropdown` changed `@height` property to use `max-height` instead of a fixed height.[(#1635)](https://github.com/hashicorp/design-system/pull/1635) Thanks [@KristinLBradley](https://github.com/KristinLBradley)! 

Reviewer: ```suggestion
- `Pagination` - fixed a bug with `aria-label` on the component. [#1627](https://github.com/hashicorp/design-system/pull/1627) Thanks [@MelSumner](https://github.com/MelSumner)! 

- `Dropdown` - changed the `@height` property to use `max-height` instead of a fixed height. [#1635](https://github.com/hashicorp/design-system/pull/1635) Thanks [@KristinLBradley](https://github.com/KristinLBradley)! 
```

## 2.12.0

> **Month Day, Year**
> 
> - `Design tokens` added color tokens for “Vault Secrets” product [(#1640)](https://github.com/hashicorp/design-system/pull/1640) Thanks [@didoo](https://github.com/didoo)! 
> 
> - `IconTile` updated component to include `vault-secrets` product option.
> 
> - `Text` - Removed leftover `console.log` from code. [(#1615)](https://github.com/hashicorp/design-system/pull/1615) Thanks [@didoo](https://github.com/didoo)! 
> 
> - `Dropdown` refactored the layout of the checkbox and radio inputs to make the gap between the inputs and the associated text, as well as the icon and count, clickable. [(#1618)](https://github.com/hashicorp/design-system/pull/1618) Thanks [@KristinLBradley](https://github.com/KristinLBradley)!
> 
> - `Hds::Link::Standalone` change font-weight from 500 to 400 to match font-weight of `Hds::Button`. [(#1617)](https://github.com/hashicorp/design-system/pull/1617) Thanks [@KristinLBradley](https://github.com/KristinLBradley)!
> 
> - `Stepper` removed some CSS declarations that were not used/applied. [(#1628)](https://github.com/hashicorp/design-system/pull/1628) Thanks [@didoo](https://github.com/didoo)!
> 
> - 🔄 Updated dependencies:
>     - @hashicorp/design-system-tokens@1.8.0
>     - @hashicorp/ember-flight-icons@3.1.2

Reviewer: ```suggestion
**September 15, 2023**

- `IconTile` - added a new variant for the `vault-secrets` product. [#1640](https://github.com/hashicorp/design-system/pull/1640) Thanks [@didoo](https://github.com/didoo)! 

- `Text` - removed leftover `console.log` from code. [#1615](https://github.com/hashicorp/design-system/pull/1615) Thanks [@didoo](https://github.com/didoo)! 

- `Dropdown` - refactored the layout of the checkbox and radio inputs to make the gap between the inputs and the associated text, as well as the icon and count, clickable. [#1618](https://github.com/hashicorp/design-system/pull/1618) Thanks [@KristinLBradley](https://github.com/KristinLBradley)!

- `Hds::Link::Standalone` - changed the font-weight from 500 to 400 to match the font-weight of `Hds::Button`. [#1617](https://github.com/hashicorp/design-system/pull/1617) Thanks [@KristinLBradley](https://github.com/KristinLBradley)!

- `Stepper` - removed some CSS declarations that were not used/applied. [#1628](https://github.com/hashicorp/design-system/pull/1628) Thanks [@didoo](https://github.com/didoo)!

- 🔄 Updated dependencies:
    - @hashicorp/design-system-tokens@1.8.0
    - @hashicorp/ember-flight-icons@3.1.2
```

## 2.11.0

> **Month Day, Year**
> 
> - `Hds::Text` added new `Text` component. [(#1490)](https://github.com/hashicorp/design-system/pull/1490) Thanks [@didoo](https://github.com/didoo)!
> 
> - `Hds::Form::MaskedInput` add `hasCopyButton` argument. [(#1587)](https://github.com/hashicorp/design-system/pull/15807) Thanks [@alex-ju](https://github.com/alex-ju)!
> 
> - `Form::Indicator` removed aria-hidden from the "optional" span [(#1577)](https://github.com/hashicorp/design-system/pull/1577) Thanks [@DingoEatingFuzz](https://github.com/DingoEatingFuzz)!
> 
> - `ember-named-blocks-polyfill` removed as all consumers of HDS are on Ember 3.25 or later now. This can be installed locally if it is still needed. [(#1606)](https://github.com/hashicorp/design-system/pull/1577) Thanks [@Dhaulagiri](https://github.com/didoo)! 
> 
> - `Alert`, `Toast` fixed an issue with anchor tag color styles within Description that had been overriding `Hds::Link color`; changed the default color for HTML links within Description to "neutral" to better align with existing guidance for links in the actions and improve accessible contrast. [(#1576)](https://github.com/hashicorp/design-system/pull/1577) Thanks [@KristinLBradley](https://github.com/KristinLBradley)! 
> 
> - 🔄 Updated dependencies:
>     - @hashicorp/ember-flight-icons@3.1.1

Reviewer: ```suggestion
**August 31, 2023**

- `Hds::Text` - added new `Text` component. [#1490](https://github.com/hashicorp/design-system/pull/1490) Thanks [@didoo](https://github.com/didoo)!

- `Hds::Form::MaskedInput` - added `hasCopyButton` argument. [#1587](https://github.com/hashicorp/design-system/pull/15807) Thanks [@alex-ju](https://github.com/alex-ju)!

- `Form::Indicator` - removed aria-hidden from the "optional" span [#1577](https://github.com/hashicorp/design-system/pull/1577) Thanks [@DingoEatingFuzz](https://github.com/DingoEatingFuzz)!

- `ember-named-blocks-polyfill` - removed as all consumers of HDS are on Ember 3.25 or later now. This can be installed locally if it is still needed. [#1606](https://github.com/hashicorp/design-system/pull/1577) Thanks [@Dhaulagiri](https://github.com/didoo)! 

- `Alert` and `Toast` - fixed the following issues with the links in the components: 
 
    - fixed an issue with the anchor tag color styles within Description that had been overriding `Hds::Link color`
    - changed the default color for HTML links within the Description to "neutral" to better align with existing guidance for links in the actions and improve accessible contrast. 

[#1576](https://github.com/hashicorp/design-system/pull/1577) Thanks [@KristinLBradley](https://github.com/KristinLBradley)! 

- 🔄 Updated dependencies:
    - @hashicorp/ember-flight-icons@3.1.1
```

## 2.10.0

> **Month Day, Year**
> 
> - `FileInput` component added. [(#1535)](https://github.com/hashicorp/design-system/pull/1535) Thanks [@KristinLBradley](https://github.com/KristinLBradley)!
> 
> - `hds/copy/index.js` removed unnecessary export. [(#1570)](https://github.com/hashicorp/design-system/pull/1570) Thanks [@Dhaulagiri](https://github.com/Dhaulagiri)! 
> 
> - `Hds::Tabs` fix missing tab indicator when used in Modal or Flyout [(#1575)](https://github.com/hashicorp/design-system/pull/1575) Thanks [@alex-ju](https://github.com/alex-ju)! 
> 
> - `Copy::Snippet` add support for container in and update API docs. [(#1567)](https://github.com/hashicorp/design-system/pull/1567) Thanks [@MelSumner](https://github.com/MelSumner)!
> 
> - `Hds::Tabs` - Fix tabs and panels misbehaving on route change. [(#1571)](https://github.com/hashicorp/design-system/pull/1571) Thanks [@alex-ju](https://github.com/alex-ju)!
> 
> - `error icon` update for copy components. [(#1568)](https://github.com/hashicorp/design-system/pull/1568) Thanks [@MelSumner](https://github.com/MelSumner)! 
> 
> - `Hds::SegmentedGroup` prevent `border-radius` from interfering with underlying elements. [(#1555)](https://github.com/hashicorp/design-system/pull/1555) Thanks [@alex-ju](https://github.com/alex-ju)! 
> 
> - `ThSort` resolved issue where ThSort was not supporting right-aligned text properly [(#1565)](https://github.com/hashicorp/design-system/pull/1565) Thanks [@MelSumner](https://github.com/MelSumner)!
> 
> - 🔄 Updated dependencies:
>     - @hashicorp/ember-flight-icons@3.1.0

Reviewer: ```suggestion
**August 15, 2023**

- `FileInput` - component added. [#1535](https://github.com/hashicorp/design-system/pull/1535) Thanks [@KristinLBradley](https://github.com/KristinLBradley)!

- Removed unnecessary export of `hds/copy/index.js`. [#1570](https://github.com/hashicorp/design-system/pull/1570) Thanks [@Dhaulagiri](https://github.com/Dhaulagiri)! 

- `Hds::Tabs` - fixed the missing tab indicator when used in Modal or Flyout [#1575](https://github.com/hashicorp/design-system/pull/1575) Thanks [@alex-ju](https://github.com/alex-ju)! 

- `Copy::Snippet` - added support for container and updated the API docs. [(#1567)](https://github.com/hashicorp/design-system/pull/1567) Thanks [@MelSumner](https://github.com/MelSumner)!

- `Hds::Tabs` - fixed tabs and panels misbehaving on route change. [(#1571)](https://github.com/hashicorp/design-system/pull/1571) Thanks [@alex-ju](https://github.com/alex-ju)!

- `Copy::Button` and `Copy::Snippet` - updated the error icon. [#1568](https://github.com/hashicorp/design-system/pull/1568) Thanks [@MelSumner](https://github.com/MelSumner)! 

- `Hds::SegmentedGroup` - prevented `border-radius` from interfering with underlying elements. [#1555](https://github.com/hashicorp/design-system/pull/1555) Thanks [@alex-ju](https://github.com/alex-ju)! 

- `ThSort` - resolved an issue where right-aligned text was not properly supported. [#1565](https://github.com/hashicorp/design-system/pull/1565) Thanks [@MelSumner](https://github.com/MelSumner)!

- 🔄 Updated dependencies:
    - @hashicorp/ember-flight-icons@3.1.0
```

## 2.9.0

> **Month Day, Year**
> 
> - `Hds::CopyButton` and `Hds::CopySnippet` components added. [(#1488)](https://github.com/hashicorp/design-system/pull/1488) Thanks [@MelSumner](https://github.com/MelSumner)!
> 
> - `copywrite headers` adjusted closing brace to avoid terminal noise [(#1539)](https://github.com/hashicorp/design-system/pull/1539) Thanks [@Dhaulagiri](https://github.com/Dhaulagiri)! 
> 
> - `Hds::Modal` reduce test flakiness around closing Modal when using `@ember/test-helpers` [(#1549)](https://github.com/hashicorp/design-system/pull/1549) Thanks [@fivetanley](https://github.com/fivetanley)! 
> 
> - `Embroider warnings` fix caused by incorrect export of internal utility functions. [(#1530)](https://github.com/hashicorp/design-system/pull/1530) Thanks [@alex-ju](https://github.com/alex-ju)! 
> 
> - `Hds::Flyout` - reduce test flakiness around closing Flyout when using `@ember/test-helpers`. [(#1552)](https://github.com/hashicorp/design-system/pull/1552) Thanks [@alex-ju](https://github.com/alex-ju)! 
> 
> - `@layout` fix error message for unexpected values. [(#1529)](https://github.com/hashicorp/design-system/pull/1529) Thanks [@alex-ju](https://github.com/alex-ju)!  
> 
> - 🔄 Updated dependencies:
>     - @hashicorp/ember-flight-icons@3.0.9

Reviewer: ```suggestion
**August 7, 2023**

- `Hds::CopyButton` and `Hds::CopySnippet` - components added. [#1488](https://github.com/hashicorp/design-system/pull/1488) Thanks [@MelSumner](https://github.com/MelSumner)!

- Adjusted closing brace on copywrite headers to avoid terminal noise. [#1539](https://github.com/hashicorp/design-system/pull/1539) Thanks [@Dhaulagiri](https://github.com/Dhaulagiri)! 

- `Hds::Modal` - reduced test flakiness around closing Modal when using `@ember/test-helpers` [#1549](https://github.com/hashicorp/design-system/pull/1549) Thanks [@fivetanley](https://github.com/fivetanley)! 

- Fix Embroider warnings caused by incorrect export of internal utility functions. [#1530](https://github.com/hashicorp/design-system/pull/1530) Thanks [@alex-ju](https://github.com/alex-ju)! 

- `Hds::Flyout` - reduced test flakiness around closing Flyout when using `@ember/test-helpers`. [#1552](https://github.com/hashicorp/design-system/pull/1552) Thanks [@alex-ju](https://github.com/alex-ju)! 

- `Hds::Form::Field` - fixed error message for unexpected `@layout` values. [#1529](https://github.com/hashicorp/design-system/pull/1529) Thanks [@alex-ju](https://github.com/alex-ju)!  

- 🔄 Updated dependencies:
    - @hashicorp/ember-flight-icons@3.0.9
```

## 2.8.1

> **Month Day, Year**
> 
> - `Hds::SideNav` - Fixed issue with links still being interactive (even if visually hidden) when the navigation is "minimized". [#1516](https://github.com/hashicorp/design-system/pull/1516) Thanks [@didoo](https://github.com/didoo)! 
> 
> - `Hds::PageHeader`, `title`, `subtitle`, and `description` fixed overflow of non-breaking text for elements. [(#1518)](https://github.com/hashicorp/design-system/pull/1518) Thanks [@didoo](https://github.com/didoo)!
> 
> - `Hds::Form::TextInput` add support for `datetime-local` type. [(#1494)](https://github.com/hashicorp/design-system/pull/1494) Thanks [@natmegs](https://github.com/natmegs)! 
> 
> - `Hds::SideNav` - updated layout styling for the `SideNav::List::Title` element. [(#1500)](https://github.com/hashicorp/design-system/pull/1500) Thanks [@didoo](https://github.com/didoo)! 
> 
> - Upgrade `ember-style-modifier` to `3.0.1`. [(#1520)](https://github.com/hashicorp/design-system/pull/1520) Thanks [@alex-ju](https://github.com/alex-ju)!
> 
> - 🔄 Updated dependencies:
>     - @hashicorp/design-system-tokens@1.7.0
>     - @hashicorp/ember-flight-icons@3.0.8

Reviewer: ```suggestion
**July 26, 2023**

- `Hds::SideNav` - fixed an issue with links still being interactive (even if visually hidden) when the navigation is "minimized". [#1516](https://github.com/hashicorp/design-system/pull/1516) Thanks [@didoo](https://github.com/didoo)! 

- `Hds::PageHeader` - fixed overflow of non-breaking text for `title`, `subtitle`, and `description` elements. [#1518](https://github.com/hashicorp/design-system/pull/1518) Thanks [@didoo](https://github.com/didoo)!

- `Hds::Form::TextInput` - added support for `datetime-local` type. [#1494](https://github.com/hashicorp/design-system/pull/1494) Thanks [@natmegs](https://github.com/natmegs)! 

- `Hds::SideNav` - updated layout styling for the `SideNav::List::Title` element. [#1500](https://github.com/hashicorp/design-system/pull/1500) Thanks [@didoo](https://github.com/didoo)! 

- Upgraded `ember-style-modifier` to `3.0.1`. [#1520)(https://github.com/hashicorp/design-system/pull/1520) Thanks [@alex-ju](https://github.com/alex-ju)!

- 🔄 Updated dependencies:
    - @hashicorp/design-system-tokens@1.7.0
    - @hashicorp/ember-flight-icons@3.0.8
```

## 2.8.0

> **Month Day, Year**
> 
> - `Hds::Card` ⛔️ **Breaking change** updated default value of @overflow argument to "visible" to address an area of consumer confusion and better support the most common use cases. 
> [(#1492)](https://github.com/hashicorp/design-system/pull/1492) Thanks [@KristinLBradley](https://github.com/KristinLBradley)! 
>     - Technically, this is a breaking change as it will require consumers relying upon the previous hidden default value to now manually set the value. The result of not setting the a hidden value can cause square edges of some images to "stick out" and overlap the rounded corners of the Card itself. We considered this to be a fairly minor, low-impact issue however which would not affect functionality or usability.
> 
> - `Hds::Form::TextInput` add loading state on "search" type. [(#1452)](https://github.com/hashicorp/design-system/pull/1492) Thanks [@alex-ju](https://github.com/alex-ju)! 
> 
> - `Hds::Form::MaskedInput` add component. [(#1468)](https://github.com/hashicorp/design-system/pull/1492) Thanks [@alex-ju](https://github.com/alex-ju)!
> 
> - `Hds::Accordion component` add component [(#1423)](https://github.com/hashicorp/design-system/pull/1423) Thanks [@KristinBradley](https://github.com/KristinLBradley)!
> 
> - `Hds::PageHeader` set position to 'relative'. [(#1466)](https://github.com/hashicorp/design-system/pull/1466) Thanks [@alex-ju](https://github.com/alex-ju)!
> 
> - `Hds::Textarea` fix border and text color for readonly state. [(#1470)](https://github.com/hashicorp/design-system/pull/1470) Thanks [@alex-ju](https://github.com/alex-ju)!
> 
> - `Hds::Modal` prevent onClose callback function invocation when isDismissDisabled is true. [(#1456)](https://github.com/hashicorp/design-system/pull/145) Thanks [@alex-ju](https://github.com/alex-ju)!
> 
> - `font-weight` of the `button` mixin explicitly to `regualr` instead of relying on inheritance (components using this mixin: `Button`, `Dropdown::ToggleButton` and soon `Accordion`) - No visual difference expected. [(#1469)](https://github.com/hashicorp/design-system/pull/1469) Thanks [@didoo](https://github.com/didoo)!
> 
> - 🔄 Updated dependencies:
>     - @hashicorp/design-system-tokens@1.6.0
>     - @hashicorp/ember-flight-icons@3.0.7

Reviewer: ```suggestion
**July 13, 2023**

- `Hds::Card` - updated default value of @overflow argument to "visible" to address an area of consumer confusion and better support the most common use cases. [#1492](https://github.com/hashicorp/design-system/pull/1492) Thanks [@KristinLBradley](https://github.com/KristinLBradley)! 

    - Technically, this is a breaking change as it will require consumers relying upon the previous hidden default value to now manually set the value. The result of not setting the a hidden value can cause square edges of some images to "stick out" and overlap the rounded corners of the Card itself. We considered this to be a fairly minor, low-impact issue however which would not affect functionality or usability.

- `Hds::Form::TextInput` - added loading state on "search" type. [#1452](https://github.com/hashicorp/design-system/pull/1492) Thanks [@alex-ju](https://github.com/alex-ju)! 

- `Hds::Form::MaskedInput` - added component. [#1468](https://github.com/hashicorp/design-system/pull/1492) Thanks [@alex-ju](https://github.com/alex-ju)!

- `Hds::Accordion` - added component [#1423](https://github.com/hashicorp/design-system/pull/1423) Thanks [@KristinBradley](https://github.com/KristinLBradley)!

- `Hds::PageHeader` - set position to 'relative'. [#1466](https://github.com/hashicorp/design-system/pull/1466) Thanks [@alex-ju](https://github.com/alex-ju)!

- `Hds::Textarea` - fixed border and text color for readonly state. [#1470](https://github.com/hashicorp/design-system/pull/1470) Thanks [@alex-ju](https://github.com/alex-ju)!

- `Hds::Modal` - prevented onClose callback function invocation when isDismissDisabled is true. [#1456](https://github.com/hashicorp/design-system/pull/145) Thanks [@alex-ju](https://github.com/alex-ju)!

- Set the `font-weight` of the `button` mixin explicitly to `regular` instead of relying on inheritance (components using this mixin: `Button`, `Dropdown::ToggleButton` and soon `Accordion`) - No visual difference expected. [#1469](https://github.com/hashicorp/design-system/pull/1469) Thanks [@didoo](https://github.com/didoo)!

- 🔄 Updated dependencies:
    - @hashicorp/design-system-tokens@1.6.0
    - @hashicorp/ember-flight-icons@3.0.7
```

## 2.7.1

> **Month Day, Year**
> 
> - `TooltipButton` added `text-align: inherit` to the "button" element. [(#1438)](https://github.com/hashicorp/design-system/pull/1438) Thanks [@didoo](https://github.com/didoo)! 
> 
> - Internal accessibility tweaks for dropdown component. [(#1444)](https://github.com/hashicorp/design-system/pull/1444) Thanks [@MelSumner](https://github.com/MelSumner)!
> 
> - **Upgraded `Ember.js` to latest stable release 4.12, including upgrades to:**
> 
>   - `ember-auto-import` from `2.6.0` to `2.6.3`
>   - `ember-cli-htmlbars` from `6.1.0` to `6.2.0`
> 
> - **Upgraded the following dependencies:**
> 
>   - `ember-focus-trap` from `1.0.1` to `1.0.2`
>   - `ember-keyboard` from `8.1.0` to `8.2.0`
>   - `ember-truth-helpers` from `3.0.0` to `3.1.1`
>   - `sass` from `1.58.3` to `1.62.1`
> 
> - Shifted our supported version of Node.js from `12.* || 14.* || >= 16` to `14.* || 16.* || >= 18`
> 
> - [#1395](https://github.com/hashicorp/design-system/pull/1395) Thanks [@alex-ju](https://github.com/alex-ju)! - 
> 
> - `Table` set `min-height` instead of `height` for the table head cells + Updated the cells' internal padding to align with the design specs in Figma. [(#1425)](https://github.com/hashicorp/design-system/pull/1425) Thanks [@didoo](https://github.com/didoo)! 
> 
> - Updated CSS code of components to use flex `gap`. [(#1433)](https://github.com/hashicorp/design-system/pull/1433) Thanks [@didoo](https://github.com/didoo)! 
> 
> - `Hds::Sidenav::Link` fixed an issue that was generating an empty node. This will lead to a minimal visual impact on some edge cases of `Alert/Toast` (multiple description items) and `SideNav` (text + generic content).
> 
> - `standalone link` style tweaks. [(#1426)](https://github.com/hashicorp/design-system/pull/1426) Thanks [@MelSumner](https://github.com/MelSumner)! 
> 
> - `Hds::Modal` and `Hds::Flyout` fix scroll management resulting in stray `style` attribute on `<body>` elementresulting. [#1434](https://github.com/hashicorp/design-system/pull/1434) Thanks [@alex-ju](https://github.com/alex-ju)!
> 
> - 🔄 Updated dependencies:
>     - @hashicorp/ember-flight-icons@3.0.6

Reviewer: ```suggestion
**June 26, 2023**

- `TooltipButton` - added `text-align: inherit` to the "button" element. [#1438](https://github.com/hashicorp/design-system/pull/1438) Thanks [@didoo](https://github.com/didoo)! 

- `Dropdown` - added internal accessibility tweaks. [#1444](https://github.com/hashicorp/design-system/pull/1444) Thanks [@MelSumner](https://github.com/MelSumner)!

- `Table` - set `min-height` instead of `height` for the table head cells and updated the cells' internal padding to align with the design specs in Figma. [#1425](https://github.com/hashicorp/design-system/pull/1425) Thanks [@didoo](https://github.com/didoo)! 

- Updated CSS code of components to use flex `gap`.

    - `Hds::Sidenav::Link` - fixed an issue that was generating an empty node. 
    - This will lead to a minimal visual impact on some edge cases of `Alert/Toast` (multiple description items) and `SideNav` (text + generic content).

 [#1433](https://github.com/hashicorp/design-system/pull/1433) Thanks [@didoo](https://github.com/didoo)! 

- Implemented minor style tweaks to: 

    - `Hds::Link::Standalone` - updated to smaller icon size
    - `Hds::Alert` and `Hds::Toast` - updated the space between actions 
    
[#1426](https://github.com/hashicorp/design-system/pull/1426) Thanks [@MelSumner](https://github.com/MelSumner)! 

- `Hds::Modal` and `Hds::Flyout` - fixed scroll management resulting in stray `style` attribute on `<body>` element. [#1434](https://github.com/hashicorp/design-system/pull/1434) Thanks [@alex-ju](https://github.com/alex-ju)!

- Upgraded `Ember.js` to latest stable release 4.12, including upgrades to:

  - `ember-auto-import` from `2.6.0` to `2.6.3`
  - `ember-cli-htmlbars` from `6.1.0` to `6.2.0`

- Upgraded the following dependencies:

  - `ember-focus-trap` from `1.0.1` to `1.0.2`
  - `ember-keyboard` from `8.1.0` to `8.2.0`
  - `ember-truth-helpers` from `3.0.0` to `3.1.1`
  - `sass` from `1.58.3` to `1.62.1`

- Shifted our supported version of Node.js from `12.* || 14.* || >= 16` to `14.* || 16.* || >= 18`

[#1395](https://github.com/hashicorp/design-system/pull/1395) Thanks [@alex-ju](https://github.com/alex-ju)!

- 🔄 Updated dependencies:
    - @hashicorp/ember-flight-icons@3.0.6
```

## 2.7.0

> **Month Day, Year**
> 
> - `Table` exposed the internal sorting properties and methods `setSortBy`, `sortBy` and `sortOrder`. [(#1421)](https://github.com/hashicorp/design-system/pull/1421) Thanks [@didoo](https://github.com/didoo)! 
> 
> - `PageHeader` component added. [(#1377)](https://github.com/hashicorp/design-system/pull/1377) Thanks [@jorytindall](https://github.com/jorytindall)!
> 
> - `Separator` component added. [(#1393)](https://github.com/hashicorp/design-system/pull/1393) Thanks [@alex-ju](https://github.com/alex-ju)! 
> 
> - `Hds::Table` - Changed the way in which the column `@width` defined by the user is applied. [(#1415)](https://github.com/hashicorp/design-system/pull/1415) Thanks [@didoo](https://github.com/didoo)!
> 
> - 🔄 Updated dependencies:
>     - @hashicorp/ember-flight-icons@3.0.5

Reviewer: ```suggestion
**June 7, 2023**

- `Hds::Table` - exposed the internal sorting properties and methods `setSortBy`, `sortBy`, and `sortOrder`. [#1421](https://github.com/hashicorp/design-system/pull/1421) Thanks [@didoo](https://github.com/didoo)! 

- `Hds::PageHeader` - added component. [#1377](https://github.com/hashicorp/design-system/pull/1377) Thanks [@jorytindall](https://github.com/jorytindall)!

- `Hds::Separator` - added component. [#1393](https://github.com/hashicorp/design-system/pull/1393) Thanks [@alex-ju](https://github.com/alex-ju)! 

- `Hds::Table` - changed the way in which the column `@width` defined by the user is applied. [#1415](https://github.com/hashicorp/design-system/pull/1415) Thanks [@didoo](https://github.com/didoo)!

- 🔄 Updated dependencies:
    - @hashicorp/ember-flight-icons@3.0.5
```

- `Stepper` - removed some CSS declarations that were not used/applied. [#1628](https://github.com/hashicorp/design-system/pull/1628) Thanks [@didoo](https://github.com/didoo)!

- 🔄 Updated dependencies:
    - @hashicorp/design-system-tokens@1.8.0
    - @hashicorp/ember-flight-icons@3.1.2

## 2.11.0

**August 31, 2023**

- `Hds::Text` - added new `Text` component. [#1490](https://github.com/hashicorp/design-system/pull/1490) Thanks [@didoo](https://github.com/didoo)!

- `Hds::Form::MaskedInput` - added `hasCopyButton` argument. [#1587](https://github.com/hashicorp/design-system/pull/15807) Thanks [@alex-ju](https://github.com/alex-ju)!

- `Form::Indicator` - removed aria-hidden from the "optional" span [#1577](https://github.com/hashicorp/design-system/pull/1577) Thanks [@DingoEatingFuzz](https://github.com/DingoEatingFuzz)!

- `ember-named-blocks-polyfill` - removed as all consumers of HDS are on Ember 3.25 or later now. This can be installed locally if it is still needed. [#1606](https://github.com/hashicorp/design-system/pull/1577) Thanks [@Dhaulagiri](https://github.com/didoo)! 

- `Alert` and `Toast` - fixed the following issues with the links in the components: 
 

>     - fixed an issue with the anchor tag color styles within Description that had been overriding `Hds::Link color`

Reviewer: ```suggestion
    - fixed an issue with the anchor tag color styles within Description that had been overriding `Hds::Link` color.
```
